src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java

```diff
@@ -102,59 +107,59 @@ private void assertMessagesAreInTransferCompleteQueue(int numberOfMessagesToBePr
     }
 
     private void sendMessagesToMq(List<RepoInPerfMessageWrapper> messagesToBeProcessed) {
-        var intervalBetweenMessagesSentToMq = getIntervalBetweenMessagesSentToMq();
+        int intervalBetweenMessagesSentToMq = getIntervalBetweenMessagesSentToMq();
         try {
-            var messageTemplate = ResourceUtility.readTestResourceFileFromEhrDirectory("small-ehr-4MB");
-            var counter = new AtomicInteger(0);
+            //
+            String messageTemplate = ResourceUtility.readTestResourceFileFromEhrDirectory("small-ehr-4MB");
+            AtomicInteger counter = new AtomicInteger(0);
             String smallEhr;
 
-            // TODO PRMT-3574 traditional for loop here that could be refactored
             for (int i = 0; i < messagesToBeProcessed.size(); i++) {
                 counter.updateAndGet(v -> v + 1);
-                String conversationId = messagesToBeProcessed.get(i).getMessage().conversationId();
-
+                String conversationId = messagesToBeProcessed.get(i).getMessage().getConversationId();
                 smallEhr = getSmallMessageWithUniqueConversationIdAndMessageId(messageTemplate, conversationId);
                 messagesToBeProcessed.get(i).start();
-
-                System.out.println("Item " + counter.get() + " - sending to mq conversationId " + conversationId);
                 inboundQueueFromMhs.sendMessage(smallEhr, conversationId);
-
                 sleepFor(intervalBetweenMessagesSentToMq);
             }
```

Comment:
Out of curiosity, why did we decide not to go for the `Iterable.forEach()` approach here again?